### DISCUSSION TITLE
How can we include both app_commands/slash commands and regular ones?

### DISCUSSION THREAD

**Discussion by nabtron** [Rapptz/discord.py#8442] [Q&A] [answered]
I currently have a bot running regular commands, it begins using:

`bot = commands.Bot(command_prefix=['xyz!'], case_insensitive=True,  intents=intents, help_command=None)`

However, the app_commands requires us to initiate the bot in an entirely different way. How can we combine both of them in the same bot?

Thanks.

**Comment by LostLuma** [comment, marked as answer]
Hello, the `Bot` and `AutoShardedBot` classes already come with a `CommandTree` attached that you can add commands to:

```py
bot = commands.Bot(...)

@bot.tree.command()
async def example(interaction: discord.Interaction) -> None:
    ...
```

Commands inside Cogs also work out of the box and are added and removed with the cog:

```py
class Example(commands.Cog):
    @discord.app_commands.command()
    async def example(self, interaction: discord.Interaction) -> None:
        ...

async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(Example())  # Commands are added to bot.tree, can be used and synced to Discord by you
```

Syncing the tree to Discord is still your responsibility when using the commands extension.

If you'd like to have the same commands available as text and slash commands there is a new `HybridCommand` type in the commands extension that creates both for you at the same time. Note that not every feature that is supported by the commands extension works with slash commands though. To use this simply define a command as you would a regular text command but use the [`hybrid_command` decorator](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.Bot.hybrid_command) instead.

**Comment by nabtron** [comment]
@LostLuma how do i define the variables like guild for slashcommand in your first code block above?

**Reply by LostLuma** [reply]
To make a command for certain guilds there is the [`@guilds` decorator](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.guilds) in which you can pass guild IDs. Please see the [full list of decorators](https://discordpy.readthedocs.io/en/stable/interactions/api.html#decorators) (and other information on this page) it should hopefully answer most of your questions.

**Comment by nabtron** [comment]
@LostLuma sorry but i don't understand how to use `copy_global_to` in the code block after `bot = commands.Bot(...)`

**Comment by nabtron** [comment]
@LostLuma  This is what i'm trying, but failing:

```
bot = commands.Bot(command_prefix=['xyz!'], case_insensitive=True,  intents=intents, help_command=None)

@bot.tree.guilds(guild=12345)
@bot.tree.command()
async def hello(interaction: discord.Interaction):
    """Says hello!"""
    await interaction.response.send_message(f'Hi, {interaction.user.mention}')
```

**Reply by LostLuma** [reply]
You'll have to be more specific in describing what is failing to receive help.

**Comment by nabtron** [comment]
@LostLuma i am unable to find this slash command in this guild, so "finding the slash command" is failing...

**Reply by LostLuma** [reply]
As written in my original reply you will need to sync the commands to Discord. You can see the [`basic example`](https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py) on how to do this (remember that the tree is already there in a `Bot` subclass). This example adds them to a single guild only, but you can leave out the `copy_global_to` and remove the `guild` argument from the `sync` call to have them added globally.

**Comment by nabtron** [comment]
@LostLuma 

I tried sync too, but still unable to see the command in the guild. 

This is the complete code, i am not sure which part has error, please help....! really stuck!
```

import discord
from discord.ext import commands
from discord import app_commands
import config

MY_GUILD = discord.Object(id=config.MY_GUILD_ID)
intents = discord.Intents.default()
intents.message_content = True
client = commands.Bot(command_prefix=['xyz!'], intents=intents)

@discord.app_commands.guilds(MY_GUILD)
@client.tree.command()
async def hello(interaction: discord.Interaction):
    """Says hello!"""
    await interaction.response.send_message(f'Hi, {interaction.user.mention}')

@client.event
async def on_ready():
  await client.tree.sync(guild=MY_GUILD)
  print('We have logged in as {0.user}'.format(client))

client.run(config.TOKEN)
```

**Reply by LostLuma** [reply]
You will have to swap the order of the `command` and `guilds` decorators, as [mentioned in the documentation](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.guilds).

**Comment by nabtron** [comment]
@LostLuma thankyou! works now! :)
